tests/services/test_service_users.py

```diff
@@ -136,3 +137,93 @@ def test_read_self(user_service, users, username):
     assert res["email"] == user.email
     # Avatar can also be seen.
     user_service.read_avatar(user.identity, user.id)
+
+
+def test_search_permissions(app, db, user_service, user_moderator, user_res):
+    """Test service search for permissions."""
+    # User can search for himself
+    search = user_service.search(
+        user_res.identity, q=f"username:{user_res._user.username}"
+    )
+    assert search.total > 0
+    # TODO user should see  moderation fields such as 'verified_at'?
+
+    # User can't search for non-confirmed users
+    with pytest.raises(PermissionDeniedError):
+        user_service.search_all(user_res.identity, user_res.id)
+
+    # Moderator can search for any user
+    search = user_service.search_all(
+        user_moderator.identity, q=f"username:{user_res._user.username}"
+    )
+    assert search.total > 0
+
+
+def test_block(app, db, user_service, user_moderator, user_res):
+    """Test user block."""
+
+    with pytest.raises(PermissionDeniedError):
+        user_service.block(user_res.identity, user_res.id)
+
+    blocked = user_service.block(user_moderator.identity, user_res.id)
+    assert blocked
+
+    ur = user_service.read(user_res.identity, user_res.id)
+    # User can't see when it was blocked
+    assert not "blocked_at" in ur.data
+    # But can see it's not active
+    assert ur.data["active"] is False
+
+    ur = user_service.read(user_moderator.identity, user_res.id)
+    # Moderator can see the blocked_at time
+    assert ur.data["blocked_at"] is not None
+
+    # TODO user is blocked, the user should not be able to search.
```

Comment:
Search permissions do not deny searches from blocked users. The permission policy for search does not contemplate the user status.